src/cs50/sql.py

```diff
@@ -29,25 +56,23 @@ def __init__(self, url, **kwargs):
             if not os.path.isfile(matches.group(1)):
                 raise RuntimeError("not a file: {}".format(matches.group(1)))
 
-            # Remember foreign_keys and remove it from kwargs
-            foreign_keys = kwargs.pop("foreign_keys", False)
+        # Create engine, disabling SQLAlchemy's own autocommit mode, raising exception if back end's module not installed
+        self._engine = sqlalchemy.create_engine(url, **kwargs).execution_options(autocommit=False)
 
-            # Create engine, raising exception if back end's module not installed
-            self.engine = sqlalchemy.create_engine(url, **kwargs)
+        # Listener for connections
+        def connect(dbapi_connection, connection_record):
 
-            # Enable foreign key constraints
-            if foreign_keys:
-                def connect(dbapi_connection, connection_record):
-                    if type(dbapi_connection) is sqlite3.Connection:  # If back end is sqlite
-                        cursor = dbapi_connection.cursor()
-                        cursor.execute("PRAGMA foreign_keys=ON")
-                        cursor.close()
-                sqlalchemy.event.listen(self.engine, "connect", connect)
+            # Disable underlying API's own emitting of BEGIN and COMMIT
+            dbapi_connection.isolation_level = None
 
-        else:
+            # Enable foreign key constraints
+            if type(dbapi_connection) is sqlite3.Connection:  # If back end is sqlite
+                cursor = dbapi_connection.cursor()
+                cursor.execute("PRAGMA foreign_keys=ON")
```

Comment:
Yeah, decided to remove, since (1) we'd added an arbitrary kwarg that's not supported by the underlying engine object (we were popping it from kwargs before instantiating the engine, which felt sloppy) and (2) pedagogically, if we're going to teach foreign keys, seems best to enforce them for correctness's sake.